pkg/nfd-worker/nfd-worker.go

```diff
@@ -271,6 +273,36 @@ func (w *nfdWorker) Run() error {
 	labelTrigger.Reset(w.config.Core.SleepInterval.Duration)
 	defer labelTrigger.Stop()
 
+	// Get pod owner reference
+	podName := os.Getenv("POD_NAME")
+	client, err := w.getKubeClient()
+	if err != nil {
+		return fmt.Errorf("failed to get kube client: %w", err)
+	}
+
+	selfPod, err := client.CoreV1().Pods(w.kubernetesNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get pod %q: %w", podName, err)
+	}
```

Comment:
I'd suggest to not error out – or at least not be this strict. To make it easy to run nfd-worker from the command line for dev purposes.

Either just log an error, or (more preferable?) do the `if podName != ""` like below.